cmd/kubeadm/app/phases/validate/validate.go

```diff
@@ -50,20 +65,17 @@ func Validate(kubeconfigPath string) error {
 		if len(nodeList.Items) < 1 {
 			return false, nil
 		}
-		n := &nodeList.Items[0]
-		if !v1.IsNodeReady(n) {
-			fmt.Println("[validate] First node has registered, but is not ready yet")
-			return false, nil
+		if waitForReadiness {
+			n := &nodeList.Items[0]
+			if !v1.IsNodeReady(n) {
+				fmt.Println("[validate] First node has registered, but is not ready yet")
+				return false, nil
+			}
 		}
 
-		fmt.Printf("[validate] First node is ready after %f seconds\n", time.Since(start).Seconds())
+		fmt.Printf("[validate] First node is registered after %f seconds\n", time.Since(start).Seconds())
```

Comment:
The message should be different in case if `waitForReadiness` is `true`.
Maybe the check above should look like
```
if !waitForReadiness {
	fmt.Printf("[validate] First node registered after %f seconds\n", time.Since(start).Seconds())
	return true, nil
}
```
and in this place, revert it back to "is ready"